### PR TITLE
First batch of new wire tests.

### PR DESCRIPTION
**Pull request by Kangz**
PTAL, this also makes it so all the unittests are in the same binary.

I'll keep making wire tests until the TODO at the end of WireTests.cpp is finished.

### CODE REVIEW

**Comment by Kangz** [comment]
PTAL, finished adding all the wire tests I wanted to add

File: src/tests/unittests/WireTests.cpp [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+        .WillOnce(Return(apiBuilder));
+
+    EXPECT_CALL(api, SamplerBuilderSetFilterMode(apiBuilder, NXT_FILTER_MODE_LINEAR, NXT_FILTER_MODE_LINEAR, NXT_FILTER_MODE_NEAREST))
+        .Times(1);
+
+    FlushClient();
+}
+
+// Test that the wire is able to send arrays of numerical values
+static uint32_t testPushConstantValues[4] = {
```
Comment (kainino0x) [inline src/tests/unittests/WireTests.cpp]: make this const or constexpr?

File: src/tests/unittests/WireTests.cpp [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+
+    FlushClient();
+
+    EXPECT_CALL(*mockBuilderErrorCallback, Call(NXT_BUILDER_ERROR_STATUS_ERROR, _, 1, 2)).Times(1);
+    EXPECT_CALL(*mockBuilderErrorCallback, Call(NXT_BUILDER_ERROR_STATUS_ERROR, _, 3, 4)).Times(1);
+
+    FlushServer();
+}
+
+// Test that we get a succes builder error status when no error happens
```
Comment (kainino0x) [inline src/tests/unittests/WireTests.cpp]: "succes"

**Comment by Kangz** [comment]
Thanks for the review!

**Pull request merged**